### ISSUE TITLE
TypeError: unhashable type: 'list' (docutils 0.17.1)

### ISSUE THREAD

**Issue by jedbrown** [mcmtroffaes/sphinxcontrib-bibtex#283]
Pinning `docutils==0.16` resolves this problem.

```
# Sphinx version: 4.4.0
# Python version: 3.10.2 (CPython)
# Docutils version: 0.17.1 release
# Jinja2 version: 3.0.3
# Last messages:

# Loaded extensions:
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/sphinx/cmd/build.py", line 280, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
  File "/usr/local/lib/python3.10/site-packages/sphinx/application.py", line 230, in __init__
    self.setup_extension(extension)
  File "/usr/local/lib/python3.10/site-packages/sphinx/application.py", line 387, in setup_extension
    self.registry.load_extension(self, extname)
  File "/usr/local/lib/python3.10/site-packages/sphinx/registry.py", line 433, in load_extension
    mod = import_module(extname)
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/usr/local/lib/python3.10/site-packages/sphinxcontrib/bibtex/__init__.py", line 9, in <module>
    from .domain import BibtexDomain
  File "/usr/local/lib/python3.10/site-packages/sphinxcontrib/bibtex/domain.py", line 26, in <module>
    import pybtex_docutils
  File "/usr/local/lib/python3.10/site-packages/pybtex_docutils/__init__.py", line 28, in <module>
    from pybtex.backends import BaseBackend
  File "/usr/local/lib/python3.10/site-packages/pybtex/backends/__init__.py", line 26, in <module>
    from pybtex.plugin import Plugin
  File "/usr/local/lib/python3.10/site-packages/pybtex/plugin/__init__.py", line 26, in <module>
    import pkg_resources
  File "/usr/local/lib/python3.10/site-packages/pkg_resources/__init__.py", line 3243, in <module>
    def _initialize_master_working_set():
  File "/usr/local/lib/python3.10/site-packages/pkg_resources/__init__.py", line 3226, in _call_aside
    f(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/pkg_resources/__init__.py", line 3255, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/local/lib/python3.10/site-packages/pkg_resources/__init__.py", line 559, in _build_master
    ws = cls()
  File "/usr/local/lib/python3.10/site-packages/pkg_resources/__init__.py", line 552, in __init__
    self.add_entry(entry)
  File "/usr/local/lib/python3.10/site-packages/pkg_resources/__init__.py", line 606, in add_entry
    self.entry_keys.setdefault(entry, [])
TypeError: unhashable type: 'list'
```

**Comment by mcmtroffaes** [comment]
I wish I could tell where this is coming from... How can we tell this is caused by sphinxcontrib-bibtex, and isn't just a docutils bug unrelated to sphinxcontrib-bibtex?

**Comment by mcmtroffaes** [comment]
I tried to reproduce this, however it works fine for me. I'm guessing this might be a problem unique to your setup, maybe a botched python installation, or something that's somehow different for you compared to my own test. Can you give me the exact steps to get to the error from a fresh virtual environment? I did the following on a fresh venv in Python 3.10.2 on windows:

```
pip install sphinxcontrib-bibtex
pip install pytest
pytest
```

and all worked.

**Comment by mcmtroffaes** [comment]
(In the above, the pytest is run from the sphinxcontrib-bibtex folder to run the regression tests.)

**Comment by mcmtroffaes** [comment]
Thanks @jeremylt. Unfortunately, I still can't reproduce... I tried the code below from the test suite from a fresh install (could add these to the regression testing if we get to the bottom of this). This is the same as running "make html" from https://github.com/mcmtroffaes/sphinxcontrib-bibtex/tree/develop/test/roots/test-sphinx after editing the ``conf.py`` file to use sphinx_rtd_theme or furo. Both tests pass without errors.

```
@pytest.mark.sphinx('html', testroot='sphinx',
                    confoverrides={
                        'html_theme': 'sphinx_rtd_theme'})
def test_sphinx_rtd_theme(app, warning) -> None:
    app.build()
    assert not warning.getvalue()


@pytest.mark.sphinx('html', testroot='sphinx',
                    confoverrides={
                        'html_theme': 'furo'})
def test_furo(app, warning) -> None:
    app.build()
    assert not warning.getvalue()
```

Here's my ``pip freeze`` for reference:

```
alabaster==0.7.12
atomicwrites==1.4.0
attrs==21.4.0
Babel==2.9.1
beautifulsoup4==4.10.0
certifi==2021.10.8
charset-normalizer==2.0.11
colorama==0.4.4
docutils==0.17.1
furo==2022.1.2
idna==3.3
imagesize==1.3.0
iniconfig==1.1.1
Jinja2==3.0.3
latexcodec==2.0.1
MarkupSafe==2.0.1
packaging==21.3
pluggy==1.0.0
py==1.11.0
pybtex==0.24.0
pybtex-docutils==1.0.1
Pygments==2.11.2
pyparsing==3.0.7
pytest==6.2.5
pytz==2021.3
PyYAML==6.0
requests==2.27.1
six==1.16.0
snowballstemmer==2.2.0
soupsieve==2.3.1
Sphinx==4.4.0
sphinx-rtd-theme==1.0.0
sphinxcontrib-applehelp==1.0.2
sphinxcontrib-bibtex==2.4.1
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==2.0.0
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.5
toml==0.10.2
urllib3==1.26.8
```

I'm sorry I can't be more helpful at this point, but I hope we can get to the bottom of this.

**Comment by jedbrown** [comment]
I'm really sorry about this. In both projects @jeremylt and I tested, the `conf.py` appended a path for breathe.
```python
import sys, breathe
sys.path.append(breathe.__path__)
```
This used to be required, but now causes lots of surprising errors. This was the first one I found, the stack trace was in sphinxcontrib-bibtex, and pinning docutils fixed it.

After your replies, I was able to reduce it to `sphinx-quickstart` with a new theme and adding the path as above. Probably the "fix" (better error message, or be a little more resilient to paths) should go upstream to docutils. In any case, thanks for your support, and sorry for taking your time with this spurious report.
